tests/appsec/test_automated_login_events.py

```diff
@@ -1348,3 +1219,771 @@ def validate_iden(meta):
+
+    def test_login_event_blocking_auto_id(self):
+        assert self.config_state_1[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        assert self.r_login.status_code == 200
+
+        assert self.config_state_2[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        assert self.config_state_3[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        interfaces.library.assert_waf_attack(self.r_login_blocked, rule="block-user-id")
+        assert self.r_login_blocked.status_code == 403
+
+    def setup_login_event_blocking_auto_login(self):
+        rc.rc_state.reset().apply()
+
+        self.config_state_1 = rc.rc_state.set_config(*CONFIG_ENABLED).apply()
+        self.r_login = weblog.post("/login?auth=local", data=login_data(context, USER, PASSWORD))
+
+        self.config_state_2 = rc.rc_state.set_config(*BLOCK_USER_RULE).apply()
+        self.config_state_3 = rc.rc_state.set_config(*BLOCK_USER_LOGIN).apply()
+        self.r_login_blocked = weblog.post("/login?auth=local", data=login_data(context, USER, PASSWORD))
+
+    def test_login_event_blocking_auto_login(self):
+        assert self.config_state_1[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        assert self.r_login.status_code == 200
+
+        assert self.config_state_2[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        assert self.config_state_3[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        interfaces.library.assert_waf_attack(self.r_login_blocked, rule="block-user-login")
+        assert self.r_login_blocked.status_code == 403
+
+    def setup_login_event_blocking_sdk(self):
+        rc.rc_state.reset().apply()
+
+        self.config_state_1 = rc.rc_state.set_config(*CONFIG_ENABLED).apply()
+        self.r_login = weblog.post(
+            "/login?auth=local&sdk_event=success&sdk_user=sdkUser", data=login_data(context, UUID_USER, PASSWORD)
+        )
+
+        self.config_state_2 = rc.rc_state.set_config(*BLOCK_USER_RULE).apply()
+        self.config_state_3 = rc.rc_state.set_config(*BLOCK_USER_ID).apply()
+        self.r_login_blocked = weblog.post(
+            "/login?auth=local&sdk_event=success&sdk_user=sdkUser", data=login_data(context, UUID_USER, PASSWORD)
+        )
+
+    @missing_feature(context.library == "nodejs", reason="SDK blocking not implemented")
+    def test_login_event_blocking_sdk(self):
+        assert self.config_state_1[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        assert self.r_login.status_code == 200
+
+        assert self.config_state_2[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        assert self.config_state_3[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        interfaces.library.assert_waf_attack(self.r_login_blocked, rule="block-user-id")
+        assert self.r_login_blocked.status_code == 403
+
+
+@rfc("https://docs.google.com/document/d/1RT38U6dTTcB-8muiYV4-aVDCsT_XrliyakjtAPyjUpw")
+@features.user_monitoring
+@scenarios.remote_config_mocked_backend_asm_dd
+class Test_Auto_User_Instrum_Mode_Capability:
```

Comment:
then you need to add that class to the manifest too right ? feel free to enable it for nodejs with the same version as the other tests :+1:

Reply:
Yep, done! thanks :smile: